### PR TITLE
Feature 607 dockermigrate

### PR DESCRIPTION
**Pull request by fentoad72**
Moved all Docker files, DOCKERFILE, and docker_*.sh script into new director ci/docker

Changed paths in travis.yml and other shell scripts under ci/travis_jobs to point to ci/docker as needed

Using 3.1 tags for data because develop is incomplete

Are we moving toward one. big datafile instead of components for 4.0 and beyond/?

Link to data caching

### CODE REVIEW

**Comment by fentoad72** [comment]
Related to #597 
Relatead to Data Caching

File: .travis.yml [hunk 1]
```diff
@@ -39,7 +39,8 @@ services:
 # NOT on the travis vm space.
 python:
   - 3.7
```
Comment (georgemccabe) [inline .travis.yml]: I would remove 3.7 from the version list so we only run with 1 version of python, unless we really need multiple versions here. Most stages/jobs explicitly list the python version, so I don't know that having an extra version is doing much at this time.

File: ci/docker/docker_setup.sh [hunk 2]
```diff
@@ -3,7 +3,9 @@ echo 'doing docker build'
 # Note: adding --build-arg <arg-name> without any value tells docker to
 #  use value from local environment (export DO_GIT_CLONE)
 
-docker build -t ${DOCKERHUB_TAG} --build-arg SOURCE_BRANCH=${DOCKERHUB_DEFAULT_TAGNAME} --build-arg MET_BRANCH=${DOCKERHUB_MET_TAGNAME} --build-arg DO_GIT_CLONE ${TRAVIS_BUILD_DIR}/internal_tests/docker
+#${TRAVIS_BUILD_DIR}/ci/docker/get_data_volumes.py
```
Comment (georgemccabe) [inline ci/docker/docker_setup.sh]: I'm not entirely sure if these docker_* scripts belong in ci/docker. They are run by Travis. I think ci/docker should contain only things used exclusively by Docker, i.e. build hooks, Dockerfile, etc.

File: ci/docker/get_data_volumes.py [hunk 3]
```diff
@@ -4,7 +4,8 @@
 import subprocess
 import shlex
 
-METPLUS_VERSION = 'develop'
+#METPLUS_VERSION = 'develop'
```
Comment (georgemccabe) [inline ci/docker/get_data_volumes.py]: We need to be sure that we update this back to develop once those are all available or else new use cases will fail.

**Review by georgemccabe — CHANGES_REQUESTED** [review]
I made a few comments inline.

We should check with @JohnHalleyGotway , but I'm not sure we want to remove the Dockerfile and build hooks for docker_data. The Dockerfile is needed to run docker build for the docker data volumes and the build hooks file I thought was going to be used to trigger the data volume creation and then create a big data volume that contains all of the subsets.

**Comment by georgemccabe** [comment]
I just pushed a change to use 'develop' as the branch version for the data volumes used in tests and added support for an additional argument to get_data_volumes.py of 'all_metplus_data.' Once the Travis builds passed, I will merge into develop and check that Docker properly triggers the builds for metplus and metplus-data.

**Pull request merged**